Libraries/Azure.psm1

```diff
@@ -1589,7 +1589,7 @@ Function Generate-AzureDeployJSONFile ($RGName, $ImageName, $osVHD, $RGXMLData,
         Add-Content -Value "$($indents[4])}," -Path $jsonFile
         #endregion
 
-        if ([string]::IsNullOrEmpty($hasWALA) -or $hasWALA -eq 'Yes') {
+        if ([string]::IsNullOrEmpty($global:hasWALA) -or $global:hasWALA -eq 'Yes') {
```

Comment:
we should avoid introducing new global variables as possible, can you pass this via parameter? 
also where does this value comes from?

Reply:
This value need to be set as part of Controller using SetGlobalVariables, so it needs to be global.

Reply:
it's not necessary to be a global variable.
It can be a property of AzureProvider, and passed to Create-AllResourceGroupDeployments as parameter in DeployVMs.
In other provider that inherits AzureProvider, you can overwrite the value of this property

Reply:
also, the value could be a Boolean, instead of a string

Reply:
Modified this commit to have custom parameter